### DISCUSSION TITLE
Modbus Bridge

### DISCUSSION THREAD

**Discussion by lennycb** [arendst/Tasmota#18448] [Q&A]
Unfortunately I can't get my setup to work.

WEMOS D1 mini
![image](https://user-images.githubusercontent.com/23618370/232868005-1fd383b2-3827-4874-876a-44ce4d96e2ee.png)

Hailege TTL to RS485
![image](https://user-images.githubusercontent.com/23618370/232867376-02153f4c-3cbe-4f0e-923f-3cf5d4bb022c.png)

Wemos runs with customized firmware.

Connected:
D1 Mini --- TTL/RS485 --- Power supply
+5V-------+5V------------+5V
Tx----------Tx
Rx----------Rx
GND-------GND----------GND

setting:
![image](https://user-images.githubusercontent.com/23618370/232869104-8e14cdfa-2b82-4ec7-93f5-96c18c197c42.png)

Log:
![image](https://user-images.githubusercontent.com/23618370/232869693-c7bc512a-f161-49b4-88c2-70e98741e6e2.png)

When transmitting, the TX LED only flashes once. then stays off. RX LED stays off.

I can't find the mistake!

**Comment by barbudor** [comment]
Have you tried the basic idea to swap the connection ?
TX for the module may means output from the module, so to be connected to Wemos RX
Just a guess, no guarantee

**Reply by lennycb** [reply]
I have now swapped Tx and Rx.
No query possible.
The LEDs now behave in the opposite way.

When transmitting, the Rx LED only flashes once. then stays off. Tx LED stays off.

**Reply by lennycb** [reply]
Have now tried again with a new module and a new WEMOS.
Can query data.

But now a new problem.
With one device it works
"functioncode": 4, "startaddress": 1

it doesn't work with another device.
"functioncode": 3, "startaddress": 10801

I have to admit that the registers on this device are stupid.
They are holding registers with addresses above 49999. (e.g. 10801+40001)

How do I get this data? I can get data with a direct RS485 to USB adapter.

Nothing comes with the WEMOS.
